game_eggs/steamcmd_servers/ark_survival_ascended/egg-ark--survival-ascended.json

```diff
@@ -0,0 +1,174 @@
+            "env_variable": "SESSION_NAME",
+            "default_value": "A Pterodactyl Hosted Server",
+            "user_viewable": true,
+            "user_editable": true,
+            "rules": "required|string",
+            "field_type": "text"
+        },
+        {
+            "name": "Auto-update server",
+            "description": "This is to enable auto-updating for servers.\r\n\r\nDefault is 1. Set to 0 to disable automatic updates",
+            "env_variable": "AUTO_UPDATE",
+            "default_value": "1",
+            "user_viewable": true,
+            "user_editable": true,
+            "rules": "required|boolean",
+            "field_type": "text"
+        },
+        {
+            "name": "Battle Eye",
+            "description": "Enable BattleEye\r\n\r\n0 to disable\r\n1 to enable\r\n\r\ndefault=\"1\"",
+            "env_variable": "BATTLE_EYE",
+            "default_value": "1",
+            "user_viewable": true,
+            "user_editable": true,
+            "rules": "required|boolean",
+            "field_type": "text"
+        },
+        {
+            "name": "App ID",
+            "description": "ARK steam app id for auto updates. Leave blank to avoid auto update.",
+            "env_variable": "SRCDS_APPID",
+            "default_value": "2430930",
+            "user_viewable": true,
+            "user_editable": false,
+            "rules": "nullable|numeric",
+            "field_type": "text"
+        },
+        {
+            "name": "Additional Arguments (PARAMS)",
+            "description": "params (?ServerPassword=...) are supported here but params need to be touching. (i.e): ?ServerPassword=MyP4SsWoRd?ServerAdminPassword=sUpRp4ssW0rd",
+            "env_variable": "ARGS_PARAMS",
+            "default_value": "",
+            "user_viewable": true,
+            "user_editable": true,
+            "rules": "nullable|string",
+            "field_type": "text"
+        },
+        {
+            "name": "Max Players",
+            "description": "Specifies the maximum amount of players able to join the server.",
+            "env_variable": "MAX_PLAYERS",
+            "default_value": "70",
+            "user_viewable": true,
+            "user_editable": true,
+            "rules": "required|numeric",
+            "field_type": "text"
+        },
+        {
+            "name": "Server Admin Password",
+            "description": "",
+            "env_variable": "ARK_ADMIN_PASSWORD",
+            "default_value": "changeme",
+            "user_viewable": true,
+            "user_editable": true,
+            "rules": "required|alpha_dash",
+            "field_type": "text"
+        },
+        {
+            "name": "Query Port",
+            "description": "ARK query port used by steam server browser and ark client server browser.",
+            "env_variable": "QUERY_PORT",
+            "default_value": "27015",
+            "user_viewable": true,
+            "user_editable": true,
```

Comment:
users can not change port, this should be vieuw only

Reply:
This can be changed. I have 2 servers on the same ip. 
![image](https://github.com/parkervcp/eggs/assets/34715358/fabd2ac5-90a5-4e7f-ae06-2e8faf1efb5e)
![image](https://github.com/parkervcp/eggs/assets/34715358/f29161d9-00ac-4067-851a-bd1f9d7cbdd8)

Reply:
Yeah it's important that the user can change the ports as needed.

Reply:
No , with pterodactyl that is not allowed. Ports must be set by the admin

Reply:
no its not. the User cant assign other Ports. only Admin can do this

Reply:
So if they got assigned a different port than the defaults they have to wait for an admin to change their server? They also might request an additional port and it might not be the same as default ports.

Reply:
the ports should be all empty and requiered. That would bei the best option

Reply:
Admins can add/remove ports, as well as allocate additional ports. If end-users want to cycle through them... for whatever reason. I do not want to put restrictions on what they can change to but, regardless, they can't use a port that's not assigned to their server.

I did set min/max so it still requires a valid port but, other than this, it should be left up to the end-users.